### PR TITLE
MAISTRA-2426 move FederationStatus into MeshFederation

### PR DESCRIPTION
**Pull request by rcernich**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -102,8 +102,8 @@ require (
 	gopkg.in/yaml.v3 v3.0.0-20200615113413-eeeca48fe776
 	helm.sh/helm/v3 v3.4.2
 	honnef.co/go/tools v0.0.1-2020.1.5 // indirect
-	istio.io/api v0.0.0-20210419172736-e076ff10ec38
-	istio.io/client-go v1.9.2
+	istio.io/api v0.0.0-20210503211644-902e709f95c4
```
Comment (jwendell) [inline go.mod:105]: this should point to commit 06ab853533326ce71d85fdfec2683d54cbe002d6

**Review by jwendell — APPROVED** [review]

**Review by knrc — APPROVED** [review]

**Pull request merged**